_pages/documentation/general_docs/building/index.md

```diff
@@ -0,0 +1,207 @@
+---
+layout: documentation
+title: Building gem5
+doc: gem5 documentation
+parent: building_extras
+permalink: /documentation/general_docs/building
+authors: Bobby R. Bruce
+---
+
+# Building gem5
+
+## Dependencies
+
+* **git** : gem5 uses git for version control
+* **gcc 4.8+**: gcc is used to compiled gem5. Version 4.8+ must be used. We do
+presently support > Version 7. (Note: [Support for gcc 4 may be dropped](
+https://gem5.atlassian.net/browse/GEM5-218), and [version >7 may be
+supposed](https://gem5.atlassian.net/browse/GEM5-194) in future releases of
+gem5.
+* SCons : gem5 uses SCons as its build environment.
+* Python 2.7+ : gem5 replies on Python development libraries (due to the
+[retirement of Python 2](
+http://pyfound.blogspot.com/2019/12/python-2-sunset.html) we are [likely to
+migrate to Python 3 in future releases of gem5](
+https://gem5.atlassian.net/browse/GEM5-275).
+* protobuf 2.1+ : The protobuf library is used for trace generation and
+playback.
+* **Boost** (Optional): The Boost library is a set of general purpose C++
+libraries. It is a necessary dependency if you wish to use the SystemC
+implementation.
+
+If compiling gem5 on Debian, Ubuntu, or related Linux distributions, you may
+install all these dependencies using APT:
+
+```
+sudo apt install build-essential git m4 scons zlib1g zlib1g-dev \
+    libprotobuf-dev protobuf-compiler libprotoc-dev libgoogle-perftools-dev \
+    python-dev python libboost-all-dev
+```
+
+## Getting the code
+
+```
+git clone https://gem5.googlesource.com/public/gem5
+```
+
+## Building with SCons
+
+gem5's build system is based on SCons, an open source build system implemented
+in Python. You can find more information about scons at <http://www.scons.org>.
+The main scons file is called SConstruct and is found in the root of the source
+tree. Additional scons files are named SConscript and are found throughout the
+tree, usually near the files they're associated with.
+
+Within the root of the gem5 directory, gem5 can be built with SCons using:
+
+```
+scons build/{ISA}/gem5.{variant}
+```
+
+where `{ISA}` is the target Instruction Set Architecture, and `{variant}`
```

Comment:
`is the target` -> `is the target (guest)`

Reply:
Done